bin/custom/concerttours/src/concerttours/facades/impl/DefaultBandFacade.java

```diff
@@ -0,0 +1,60 @@
+package concerttours.facades.impl;
+
+import concerttours.data.BandData;
+import concerttours.enums.MusicType;
+import concerttours.facades.BandFacade;
+import concerttours.model.BandModel;
+import concerttours.service.BandService;
+import de.hybris.platform.servicelayer.dto.converter.Converter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DefaultBandFacade implements BandFacade
+{
+    private BandService bandService;
+
+    private Converter<BandModel, BandData> bandConverter;
+
+    public void setBandService(final BandService bandService)
+    {
+        this.bandService = bandService;
+    }
+
+    public void setBandConverter(Converter<BandModel, BandData> bandConverter) {
+        this.bandConverter = bandConverter;
+    }
+
+    @Override
+    public List<BandData> getBands()
+    {
+        return bandService.getBands().stream()
+                .map(bandModel -> bandConverter.convert(bandModel))
+                .collect(Collectors.toList());
+    }
+    @Override
+    public BandData getBand(final String name)
+    {
+        if (name == null)
+        {
+            throw new IllegalArgumentException("Band name cannot be null");
+        }
+        final BandModel band = bandService.getBandForCode(name);
+        if (band == null)
+        {
+            return null;
+        }
+
+        final List<String> genres = new ArrayList<>();
```

Comment:
CollectionUtils.emptyIfNull + stream + map + collect.
Also, this should be done in a separate populator which should be included in the band converter's populators list. So that in this class you'll just have to do bandConverter.convert(model) and this will already return a fully populated DTO.